src/ShadowDOM/wrappers/Range.js

```diff
@@ -20,23 +20,69 @@
 
   var OriginalRange = window.Range;
 
+  var ShadowRoot = scope.wrappers.ShadowRoot;
+
+  function isElement(node){
+    return node && node.nodeType == Node.ELEMENT_NODE
+  }
+
+  function getHost(node){
+    while(node && node.parentNode){
+      if (node.parentNode && node.parentNode instanceof ShadowRoot){
+        return node.parentNode.host;
+      }
+      node=node.parentNode;
+    }
+  }
+
+  function hostNodeToShadowNode(refNode,offset){
+    if (refNode.shadowRoot){
+      // Note: if the refNode is an element, then selecting a range with and offset
+      // equal to refNode.childNodes.length+1 is valid. That is why calling Math.min
+      // is necessary to make sure we select valid children.
+      var child = refNode.childNodes[Math.min(refNode.childNodes.length-1, offset)];
+      if (child){
+        var insertionPoint = ShadowDOMPolyfill.getDestinationInsertionPoints(child);
+        if (insertionPoint.length>0){
+          var parentNode = insertionPoint[0].parentNode;
+          if (isElement(parentNode)){
```

Comment:
could this be `parentNode.nodeType == Node.ELEMENT_NODE` or maybe `parentNode instanceof Element`? in other words, do you need the function?